script2.js

```diff
@@ -0,0 +1,704 @@
+    var playerRank = playerObject.totalRank;
+    if (playerRank > highestRank) {
+      highestRank = playerRank;
+      winner.push(playerObject);
+    }
+    index += 1;
+  }
+  // Output the winner if there is one
+  if (winner.length > 0) {
+    var nameOfWinner = winner[0].name;
+    return nameOfWinner;
+  }
+  // otherwise, return empty string
+  return "";
+};
+
+// Check who tied
+
+// Function to reset game conditions for the next round
+var resetGame = function () {
+  // reset the numberOfHits, numberOfRounds, currentPlayer, playerCardsArray, dealerCardsArray & list of cards, playerArray
+  numberOfHits_player = 0;
+  numberOfHits_dealer = 0;
+  numberOfRounds = 0;
+  currentPlayer = 1;
+
+  listOfCards_player = "";
+  listOfCards_dealer = "";
+
+  dealerCardsArray = [];
+  playerArray = [];
+};
+
+//// MAIN FUNCTION ////
+var main = function (input) {
+  // Default output value
+  var myOutputValue = "";
+  console.log("Game mode:");
+  console.log(gameMode);
+
+  // If gameMode is GAME_MODE_WELCOME, change gameMode to GAME_MODE_NUMBER_OF_PLAYERS and output welcome message
+  if (gameMode == GAME_MODE_WELCOME) {
+    // Change gameMode to GAME_MODE_NUMBER_OF_PLAYERS
+    gameMode = GAME_MODE_NUMBER_OF_PLAYERS;
+    console.log("Game mode:");
+    console.log(gameMode);
+    // Create welcome message
+    myOutputValue = `Welcome to BlackJack! A maximum of 4 players can play against the computer. The computer will be the dealer. Press submit for the next step.`;
+    console.log("myOutputValue");
+    console.log(myOutputValue);
+    return myOutputValue;
+  }
+
+  // If gameMode is GAME_MODE_NUMBER_OF_PLAYERS, create player object variables and change gameMode to GAME_MODE_NAMES to prompt user to enter player names
+  if (gameMode == GAME_MODE_NUMBER_OF_PLAYERS) {
+    if (
+      input == "" ||
+      input.charAt(input.length - 1) == " " ||
+      isNaN(input) == true
+    ) {
+      console.log("Invalid input. Prompt user to enter number of players");
+      myOutputValue =
+        "Please enter the number of players playing this game. A maximum of 4 players is allowed.";
+      return myOutputValue;
+    }
+
+    // store input as the global variable numberOfPlayers
+    console.log("number of players entered");
+    console.log(input);
+    numberOfPlayers = input;
+
+    // if this is not the first round, make sure to empty previous round's cards from players' hands and the totalRank
+    if (numberOfRounds > 0) {
+      var index = 0;
+      while (index < numberOfPlayers) {
+        playerArray[index].cardsArray = [];
+        playerArray[index].totalRank = 0;
+        index += 1;
+      }
+    }
+    // Create n objects of players, each with the attributes: name, playerNumber, cardsArray, totalRank, wins
+    var index = 0;
+    while (index < numberOfPlayers) {
+      var player = {
+        name: "",
+        playerNumber: index + 1,
+        cardsArray: [],
+        totalRank: 0,
+        wins: 0,
+      };
```

Comment:
Nice work using objects here for both players and computer! :) 

Perhaps this could be made into a function